CodePurpose/political_leaning_api
# add absinthe, user table, and test schema

## Describe Changes
I added absinthe to the application, a simple users table, and a seed.
This is just an initial add to get some of the pieces in place for us to iterate on. 
To better understand what is happening here checkout https://www.howtographql.com/graphql-elixir/1-getting-started/ to https://www.howtographql.com/graphql-elixir/2-queries/. This implements our changes for users roughly following that guide.
## How do I validate changes?
- run `mix deps.get`
- run `mix ecto.reset` (assumes you have setup postgres locally)
- start the server with `mix phx.server`
- download insomnia https://insomnia.rest/download
- create a request in insomnia and paste the following curl command into the URL
```
curl --request POST \
  --url http://localhost:4000/graphql \
  --header 'content-type: application/json' \
  --data '{"query":"{\n  allUsers {\n    id\n    username\n  }\n}\n"}'
```
- Your insomnia should look like the following ![image](https://user-images.githubusercontent.com/1214752/70872766-7e57d780-1f67-11ea-99e9-97c1297679b0.png)
- execute the request and validate the response is in the form
```
{
  "data": {
    "allUsers": [
      {
        "id": "2",
        "username": "IAMAUSERNAME"
      }
    ]
  }
}
```
